libPhoneNumber/NBPhoneNumberUtil.m

```diff
@@ -2622,6 +2622,68 @@ - (BOOL)isPossibleNumber:(NBPhoneNumber*)number
 }
 
 
+- (NBEValidationResult)validateNumberLength:(NSString *)number metadata:(NBPhoneMetaData *)metadata {
+    return [self validateNumberLength:number metadata:metadata type:NBEPhoneNumberTypeUNKNOWN];
+}
+
+- (NBEValidationResult)validateNumberLength:(NSString *)number metadata:(NBPhoneMetaData *)metadata type:(NBEPhoneNumberType)type {
+    NBPhoneNumberDesc *descForType = [self getNumberDescByType:metadata type:type];
+    
+    NSArray<NSNumber *> *possibleLengths = [descForType.possibleLength count] == 0 ? metadata.generalDesc.possibleLength : descForType.possibleLength;
+    
+    NSArray<NSNumber *> *localLengths = descForType.possibleLengthLocalOnly;
+    
+    if (type == NBEPhoneNumberTypeFIXED_LINE_OR_MOBILE) {
+        if ([self descHasPossibleNumberData:[self getNumberDescByType:metadata type:NBEPhoneNumberTypeFIXED_LINE]]) {
+            return [self validateNumberLength:number metadata:metadata type:NBEPhoneNumberTypeMOBILE];
+        } else {
+            NBPhoneNumberDesc *mobileDesc = [self getNumberDescByType:metadata type:NBEPhoneNumberTypeMOBILE];
+            if([self descHasPossibleNumberData:mobileDesc]) {
+                NSArray *combinedArray = [possibleLengths arrayByAddingObjectsFromArray:[mobileDesc.possibleLength count] == 0 ? metadata.generalDesc.possibleLength : mobileDesc.possibleLength];
+                possibleLengths = [combinedArray sortedArrayUsingComparator:^NSComparisonResult(NSNumber * _Nonnull obj1, NSNumber * _Nonnull obj2) {
```

Comment:
does possibleLengths = [anArray sortedArrayUsingSelector:@selector(compare:)] work instead of the whole block? (for below as well)